src/bb/fs/fsdir.c

```diff
@@ -0,0 +1,52 @@
+#include "PR/os_internal.h"
+#include "PR/bbfs.h"
+
+s32 osBbFReadDir(OSBbDirEnt* dir, u32 count) {
+    s32 rv;
+    s32 i;
+    s32 j;
+    BbFat16* fat;
+    OSBbDirEnt* d;
+
+    d = dir;
+    rv = __osBbFsGetAccess();
+
+    if (rv < 0) {
+        return rv;
+    }
+
+    fat = __osBbFat;
+
+    for (i = 0, j = 0; i < BB_INODE16_NUM; i++) {
+        unsigned char* var_s2 = fat->inode[i].name;
+        s32 k;
+
+        if (fat->inode[i].type == 0) {
+            continue;
+        }
+
+        rv++;
+        if ((d != NULL) && (j < count)) {
```

Comment:
```suggestion
        if (d != NULL && j < count) {
```
imo parens should only be used where necessary or when operator precedence is unclear. Here it is clear that comparisons come before logical operators.